multicluster/service-mirror/cluster_watcher.go

```diff
@@ -1232,7 +1241,14 @@ func (rcsw *RemoteClusterServiceWatcher) isExported(l map[string]string) bool {
 	return selector.Matches(labels.Set(l)) || remoteDiscoverySelector.Matches(labels.Set(l))
 }
 
-func (rcsw *RemoteClusterServiceWatcher) isRemoteDiscovery(svc *corev1.Service) bool {
+// isRemoteDiscovery returns true if the passed labels correspond to a local
+// mirror Service or a remote Service/Endpoints in remote-discovery mode
+func (rcsw *RemoteClusterServiceWatcher) isRemoteDiscovery(l map[string]string) bool {
```

Comment:
Let's not overload this function to deal with both exported services and mirror services/endpoints.  These are two different things and we shouldn't conflate them.